### PR TITLE
Make sure add-to-app build bundle from outer xcodebuild/gradlew sends analytics

### PR DESCRIPTION
**Pull request by xster**
## Description

Script-triggered re-entrant flutter build bundle calls were broken by https://github.com/flutter/flutter/pull/34179. Add them back. Add tests (though module_test_ios doesn't run on CI yet T.T https://github.com/flutter/flutter/issues/36120)

## Related Issues

https://github.com/flutter/flutter/issues/32864

## Tests

I added the following tests:

Integration test on Android/iOS

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Please ignore `codecov/patch`, I've fixed it at master to not have an absurd requirement for diffs.

File: packages/flutter_tools/lib/src/base/utils.dart [hunk 1]
```diff
@@ -27,6 +27,7 @@ class BotDetector {
 
         // Set by the IDEs to the IDE name, so a strong signal that this is not a bot.
         || platform.environment.containsKey('FLUTTER_HOST')
```
Comment (blasten) [inline packages/flutter_tools/lib/src/base/utils.dart]: nit: why not reusing `FLUTTER_HOST`. The comment above indicates that it's used by IDEs.

File: packages/flutter_tools/lib/src/usage.dart [hunk 2]
```diff
@@ -59,8 +59,17 @@ class Usage {
...
+    final String logFilePath = platform.environment['FLUTTER_ANALYTICS_LOG_FILE'];
+
+    _analytics = logFilePath == null ?
+        AnalyticsIO(
+          _kFlutterUA,
+          settingsName,
+          version,
+          documentDirectory: configDirOverride != null ? fs.directory(configDirOverride) : null,
+        ) :
+        LogToFileAnalytics(logFilePath);
```
Comment (blasten) [inline packages/flutter_tools/lib/src/usage.dart]: nit: maybe add a comment, so it's clear that this is used for testing only.

**Review by jmagman — APPROVED** [review]
Looks good, assuming you address @blasten's comments.

File: packages/flutter_tools/lib/src/base/utils.dart (hunk 1 above)
Comment (xster) [reply packages/flutter_tools/lib/src/base/utils.dart]: Gonna let https://github.com/flutter/flutter/pull/36208 merge first, then this PR will just be tests

**Pull request merged**